crates/emitter/src/emitter.rs

```diff
@@ -280,22 +273,13 @@ impl InstructionWriter {
         self.emit_i32(opcode, offset);
     }
 
-    pub fn get_atom_index(&mut self, value: &str) -> AtomIndex {
-        // Eventually we should be fancy and make this O(1)
-        for (i, string) in self.strings.iter().enumerate() {
-            if string == value {
-                return AtomIndex::new(i as u32);
-            }
-        }
-
-        let index = self.strings.len();
-        self.strings.push(value.to_string());
-        AtomIndex::new(index as u32)
+    pub fn get_atom_index(&mut self, value: AtomSetIndex) -> AtomIndex {
+        self.atoms.insert(value)
```

Comment:
`AtomSetIndex` and `AtomIndex` are quite confusing. Why do we have 2 type? If I understand correctly, the `AtomIndex` is the index of an `AtomSetIndex` in an `AtomList`, and the `AtomSetIndex` is the index of the entry in an `AtomSet<'alloc>`.

Should these be renamed to `SourceAtomSet<'alloc>` and `ScriptAtomSet` with there respective indexes type?

Reply:
yes, that's correct, and they need better name.
thank you for nice naming idea :D